tests/s2_inference/test_generic_clip_model.py

```diff
@@ -0,0 +1,328 @@
+            index_name=self.index_name_1, config=self.config,
+            index_settings={
+                "index_defaults": {
+                    'model': model_name,
+                    'type' : "clip"
+                }
+            }
+        )
+
+        model_properties = None
+
+        self.assertRaises(UnknownModelError, _validate_model_properties, model_name, model_properties)
+
+
+    def test_vectorise_generic_openai_clip_encode_image_results(self):
+
+        epsilon = 1e-7
+
+        image = "https://raw.githubusercontent.com/marqo-ai/marqo-clip-onnx/main/examples/coco.jpg"
+
+        model_name = "test-model"
+        model_properties = {
+                            "name": "ViT-B/32",
+                            "dimensions": 512,
+                            "url": "https://openaipublic.azureedge.net/clip/models/40d365715913c9da98579312b702a82c18be219cc2a73407c4526f58eba950af/ViT-B-32.pt",
+                            "type": "clip",
+                            }
+
+        a = vectorise(model_name, content = image, model_properties = model_properties)
+        b = vectorise("ViT-B/32", content = image)
+
+        assert np.abs(np.array(a) - np.array(b)).sum() < epsilon
+
+
+    def test_vectorise_generic_openai_clip_encode_text_results(self):
+
+        epsilon = 1e-7
+        text = "this is a test to test the custom clip output results"
+
+        model_name = "test-model"
+        model_properties = {
+                            "name": "ViT-B/32",
+                            "dimensions": 512,
+                            "url": "https://openaipublic.azureedge.net/clip/models/40d365715913c9da98579312b702a82c18be219cc2a73407c4526f58eba950af/ViT-B-32.pt",
+                            "type": "clip",
+                            }
+
+        a = vectorise(model_name, content=text, model_properties=model_properties)
+        b = vectorise("ViT-B/32", content=text)
+
+        assert np.abs(np.array(a) - np.array(b)).sum() < epsilon
+
+
+    def test_vectorise_generic_open_clip_encode_image_results(self):
+
+        epsilon = 1e-7
+
+        image = "https://raw.githubusercontent.com/marqo-ai/marqo-clip-onnx/main/examples/coco.jpg"
+
+        model_name = "test-model"
+        model_properties = {
+                            "name": "ViT-B-32-quickgelu",
+                            "dimensions": 512,
+                            "url": "https://github.com/mlfoundations/open_clip/releases/download/v0.2-weights/vit_b_32-quickgelu-laion400m_e31-d867053b.pt",
+                            "type": "open_clip",
+                            "jit" : False
+                            }
+
+        a = vectorise(model_name, content = image, model_properties = model_properties)
+        b = vectorise("open_clip/ViT-B-32-quickgelu/laion400m_e31", content = image)
+
+        assert np.abs(np.array(a) - np.array(b)).sum() < epsilon
+
+
+    def test_vectorise_generic_open_clip_encode_text_results(self):
+        epsilon = 1e-7
```

Comment:
Is it worth doing the inverse of this type of test?

That a different custom model != default clip model?